www/__tests__/notifScheduler.test.ts

```diff
@@ -0,0 +1,453 @@
+        val: DateTime.fromMillis(mockNotifs[2].trigger.at).toFormat('t'),
+      },
+      {
+        key: DateTime.fromMillis(mockNotifs[3].trigger.at).toFormat('DDD'),
+        val: DateTime.fromMillis(mockNotifs[3].trigger.at).toFormat('t'),
+      },
+      {
+        key: DateTime.fromMillis(mockNotifs[4].trigger.at).toFormat('DDD'),
+        val: DateTime.fromMillis(mockNotifs[4].trigger.at).toFormat('t'),
+      },
+    ];
+
+    // mock the cordova plugin
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'getScheduled')
+      .mockImplementation((callback) => callback(mockNotifs));
+    // call the funciton
+    const scheduledNotifs = await getScheduledNotifs(isScheduling, Promise.resolve());
+
+    expect(scheduledNotifs).toEqual(expectedResult);
+  });
+});
+
+describe('updateScheduledNotifs', () => {
+  afterEach(() => {
+    jest.restoreAllMocks(); // Restore mocked functions after each test
+  });
+
+  beforeEach(() => {
+    // mock the getUser function
+    mockGetUser.mockImplementation(() =>
+      Promise.resolve({
+        // These values are **important**...
+        //   reminder_assignment: must match a key from the reminder scheme above,
+        //   reminder_join_date: must match the first day of the mocked notifs below in the tests,
+        //   reminder_time_of_day: must match the defaultTime from the chosen reminder_assignment in the reminder scheme above
+        reminder_assignment: 'weekly',
+        reminder_join_date: '2023-11-14',
+        reminder_time_of_day: '21:00',
+      }),
+    );
+  });
+
+  it('should resolve after scheduling notifications', async () => {
+    // updateScheduleNotifs arguments
+    const reminderSchemes: any = exampleReminderSchemes;
+    let isScheduling: boolean = false;
+    const setIsScheduling: Function = jest.fn((val: boolean) => (isScheduling = val));
+    const scheduledPromise: Promise<any> = Promise.resolve();
+    // create an empty array of mock notifs from cordova plugin
+    const mockNotifs = [];
+
+    // mock the cordova plugin
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'getScheduled')
+      .mockImplementation((callback) => callback(mockNotifs));
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'cancelAll')
+      .mockImplementation((callback) => callback());
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'schedule')
+      .mockImplementation((arg, callback) => callback(arg));
+    // call the function
+    await updateScheduledNotifs(reminderSchemes, isScheduling, setIsScheduling, scheduledPromise);
+
+    expect(setIsScheduling).toHaveBeenCalledWith(true);
+    expect(logDebug).toHaveBeenCalledWith('After cancelling, there are no scheduled notifications');
+    expect(logDebug).toHaveBeenCalledWith('After scheduling, there are no scheduled notifications');
+    expect(setIsScheduling).toHaveBeenCalledWith(false);
```

Comment:
The tests should be checking the **functionality** of `notifScheduler`. In this instance we want to see that the appropriate notifications are scheduled. The `logDebug` statements are good for dev purposes but I don't think they should be the focus of our testing strategy.

After calling `updateScheduledNotifs(...)` we should verify that the function did what it was supposed to do.
In this case, I think the best way to do that is to then call `getScheduledNotifs(...)` and check the result.

Reply:
I added the test you suggested, to test `getScheduledNotifs(...)` in https://github.com/e-mission/e-mission-phone/pull/1092/commits/ec9729de081000fcdc9bed288b8da5c0ee3b77c8

Great suggestion, thanks!